custom_components/waste_collection_schedule/waste_collection_schedule/source/canadabay_nsw_gov_au.py

```diff
@@ -0,0 +1,127 @@
+import json
+from datetime import date, timedelta
+
+import requests
+from waste_collection_schedule import Collection  # type: ignore[attr-defined]
+
+TITLE = "City of Canada Bay Council"
+DESCRIPTION = "Source for City of Canada Bay Council rubbish collection."
+URL = "https://www.canadabay.nsw.gov.au/residents/waste-and-recycling/my-bins/my-bin-collection"
+TEST_CASES = {
+    "Harry's Shed": {
+        "suburb": "Concord",
+        "street_name": "Gipps Street",
+        "street_number": "1A",
+    },
+    "Five Dock Library": {
+        "suburb": "Five Dock",
+        "street_name": "Garfield Street",
+        "street_number": "4-12",
+    },
+    "Dazed cafe": {
+        "suburb": "Mortlake",
+        "street_name": "Tennyson Road",
+        "street_number": "76",
+    },
+}
+
+HEADERS = {"user-agent": "Mozilla/5.0"}
+
+
+class Source:
+    def __init__(self, suburb, street_name, street_number):
+        self.suburb = suburb
+        self.street_name = street_name
+        self.street_number = street_number
+
+    def fetch(self):
+
+        suburb_id = 0
+        street_id = 0
+        property_id = 0
+        today = date.today()
+        nextmonth = today + timedelta(30)
+
+        # Retrieve suburbs
+        r = requests.get(
+            "https://canada-bay.waste-info.com.au/api/v1/localities.json", headers=HEADERS
+        )
+        data = json.loads(r.text)
+
+        # Find the ID for our suburb
+        for item in data["localities"]:
+            if item["name"] == self.suburb:
+                suburb_id = item["id"]
+                break
+
+        if suburb_id == 0:
+            return []
+
+        # Retrieve the streets in our suburb
+        r = requests.get(
+            f"https://canada-bay.waste-info.com.au/api/v1/streets.json?locality={suburb_id}",
```

Comment:
Please pass URL parameters using the `params` argument instead of string concatenation: https://requests.readthedocs.io/en/latest/user/quickstart/#passing-parameters-in-urls

Same is true for the following requests.

Reply:
it was copypasta from Inner West Council (https://github.com/mampfes/hacs_waste_collection_schedule/pull/320) and Brisbane City Council (https://github.com/mampfes/hacs_waste_collection_schedule/pull/34). i've cleaned it up properly now.